client/components/pie-chart/legend.js

```diff
@@ -0,0 +1,63 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+
+/**
+ * Internal dependencies
+ */
+import DataType from './data/type';
+import LegendItem from './legend-item';
+import { sortDataAndAssignSections, isDataEqual } from './data';
+
+class PieChartLegend extends Component {
+	static propTypes = {
+		data: PropTypes.arrayOf( DataType ).isRequired,
+	};
+
+	constructor( props ) {
+		super( props );
+
+		this.state = this.processData( props.data );
+	}
+
+	componentWillReceiveProps( nextProps ) {
+		if ( ! isDataEqual( this.props.data, nextProps.data ) ) {
+			this.setState( this.processData( nextProps.data ) );
+		}
+	}
+
+	processData( data ) {
+		const sortedData = sortDataAndAssignSections( data );
+
+		return {
+			data: sortedData,
+			dataTotal: sortedData.reduce( ( total, datum ) => total + datum.value, 0 ),
```

Comment:
You could use https://lodash.com/docs#sumBy here :smile:

Reply:
I was on the fence about that one and ended up thinking that `reduce` was just as good, but it is not as  clear as `sumBy` would be.